code/computational_geometry/src/bresenham_line/Bresenham_Line_Algorithm.java

```diff
@@ -0,0 +1,139 @@
+package applet;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.awt.event.MouseMotionListener;
+
+public class bresenhamLine extends java.applet.Applet implements MouseListener, MouseMotionListener {
+    private static final long serialVersionUID = 1L;
+    int width;
+    int height;
+    int xa = 0,
+    ya = 0,
+    xb = 0,
+    yb = 0,
+    pixelsize = 2;
+
+    public void init() {
+        width = getSize().width;
+        height = getSize().height;
+        this.addMouseListener(this);
+        this.addMouseMotionListener(this);
+        }
+
+    // draw a line from point xa,ya into xb,yb
+    public void bresenham_line(int xa, int ya, int xb, int yb) 
+    {
+        // if point xa, ya is on the right side of point xb, yb, change them
+        if ((xa - xb) > 0) {
+            bresenham_line(xb, yb, xa, ya);
+            return;
+        }
+
+        // test inclination of line
+        // function Math.abs(y) defines absolute value y
+        if (Math.abs(yb - ya) > Math.abs(xb - xa)) {
+            // line and y axis angle is less then 45 degrees
+            // that swhy go on the next procedure
+            bresteepline(ya, xa, yb, xb);
+            return;
+        }
+
+        // line and x axis angle is less then 45 degrees, so x is guiding
+        // auxiliary variables
+        int x = xa,
+        y = ya,
+        sum = xb - xa,
+        Dx = 2 * (xb - xa),
+        Dy = Math.abs(2 * (yb - ya));
+        int prirastokDy = ((yb - ya) > 0) ? 1 : -1;
+        // draw line
+        for (int i = 0; i <= xb - xa; i++) 
+        {
+            setpix(x, y);
+            x++;
+            sum -= Dy;
+            if (sum<0) {
+                y += prirastokDy;
+                sum += Dx;
+            }
+        }
+
+    }
+
+    public void bresteepline(int xc, int yc, int xd, int yd) {   
+        // if point x3, y3 is on the right side of point x4, y4, change them
+        if ((xc - xd) > 0) {
+            bresteepline(xd, yd, xc, yc);
+            return;
+        }
+
+        int x = x3,
```

Comment:
Please add some comments. Somewhat hard to understand, especially since variable names aren't ideal.